crypto/threads_pthread.c

```diff
@@ -593,13 +593,9 @@ int ossl_rcu_call(CRYPTO_RCU_LOCK *lock, rcu_cb_fn cb, void *data)
 
     new->data = data;
     new->fn = cb;
-    /*
-     * Use __ATOMIC_ACQ_REL here to indicate that any prior writes to this
-     * list are visible to us prior to reading, and publish the new value
-     * immediately
-     */
-    new->next = ATOMIC_EXCHANGE_N(prcu_cb_item, &lock->cb_items, new,
-                                  __ATOMIC_ACQ_REL);
+
```

Comment:
I'd add a comment that says that this function must be called with write_lock mutex already held by the caller

Reply:
BTW: please remove the prcu_cb_item typedef and the
IMPL_fallback_atomic_exchange_n(prcu_cb_item) as well.